### PR TITLE
build(deps): Update sbt-buildinfo from 0.7.0 to 0.11.0

### PR DESCRIPTION
**Pull request by bloopoid**
Updates [com.eed3si9n:sbt-buildinfo](https://github.com/sbt/sbt-buildinfo) from 0.7.0 to 0.11.0.
[GitHub Release Notes](https://github.com/sbt/sbt-buildinfo/releases/tag/v0.11.0) - [Version Diff](https://github.com/sbt/sbt-buildinfo/compare/v0.7.0...v0.11.0)

I'll automatically update this PR to resolve conflicts as long as you don't change it yourself.

If you'd like to skip this version, you can just close this PR. If you have any feedback, just mention me in the comments below.

Configure Scala Steward for your repository with a [`.scala-steward.conf`](https://github.com/scala-steward-org/scala-steward/blob/e93966d6d80e4ff1b31567984aa7e70e11cb2070/docs/repo-specific-configuration.md) file.

Have a fantastic day writing Scala!

<details>
<summary>Files still referring to the old version number</summary>

The following files still refer to the old version number (0.7.0).
You might want to review and update them manually.
```
project/build.sbt
website/yarn.lock
```
</details>
<details>
<summary>Adjust future updates</summary>

Add this to your `.scala-steward.conf` file to ignore future updates of this dependency:
```
updates.ignore = [ { groupId = "com.eed3si9n", artifactId = "sbt-buildinfo" } ]
```
Or, add this to slow down future updates of this dependency:
```
dependencyOverrides = [{
  pullRequests = { frequency = "@monthly" },
  dependency = { groupId = "com.eed3si9n", artifactId = "sbt-buildinfo" }
}]
```
</details>

labels: sbt-plugin-update, early-semver-major, semver-spec-minor, old-version-remains, commit-count:1

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -304,7 +307,18 @@ lazy val frontend: Project = project
...
-    buildInfoKeys := bloopInfoKeys(nativeBridge04, jsBridge06, jsBridge1),
+    buildInfoKeys := List[BuildInfoKey](
+      Keys.organization,
+      build.BuildKeys.bloopName,
+      Keys.version,
+      Keys.scalaVersion,
+      nailgunClientLocation,
+      "zincVersion" -> Dependencies.zincVersion,
+      "bspVersion" -> Dependencies.bspVersion,
+      "nativeBridge04" -> (nativeBridge04Name + "_" + Keys.scalaBinaryVersion.value),
```
Comment (tgodzik) [inline build.sbt:318]: Previously,  we would calculate the artifact name using sbt, but this started to cause cycles when trying to make it work under new buildinfo. Instead I decided to remove everything that is not used and just figure out the artifact name using a small heuristic.

File: build.sbt [hunk 2]
```diff
@@ -707,29 +721,31 @@ val docs = project
     }
   )
 
+val jsBridge06Name = "bloop-js-bridge-0-6"
```
Comment (tgodzik) [inline build.sbt:724]: The name was actually `bloop-js-bridge-0.6`, but the `.` would get translated to `-`

File: build.sbt (hunk 1 above)
Comment (ckipp01) [reply build.sbt:318]: Huh, I would have thought there was a `Key` that could give this to you, but I can't see one either.

**Review by ckipp01 — APPROVED** [review]
Yay for removing code 👍🏼

File: build.sbt (hunk 1 above)
Comment (tgodzik) [reply build.sbt:318]: yeah, I couldn't find it either, so decided on a heuristic

**Pull request merged**